lib/utils/metrics.js

```diff
@@ -0,0 +1,146 @@
+'use strict';
+
+const assert = require('assert');
+const uuid = require('uuid');
+
+/** @typedef { import('bmutex').Lock } Lock */
+
+const metrics = exports;
+
+/**
+ * @typedef {Object} Honey
+ */
+
+class Event {
+  /**
+   *
+   * @param {string} name the name for the event
+   * @param {Event | Honey} parent a parent Event or a libhoney instance
```

Comment:
Why not `Event | null`? Is this a syntactic convenience thing? Overloading the second parameter to provide the honey instance when an event is at the top of the tree, but just using the parent's honey instance otherwise?

Reply:
Yeah, basically. The thought was it lets us easily extend "upward" passing an event where honey should've been. Not set in it